requests_mv_integrations/request_mv_integration.py

```diff
@@ -0,0 +1,1774 @@
+
+# @brief Request with retry class for TUNE Multiverse classes
+#
+# @namespace requests_mv_integrations.RequestMvIntegration
+class RequestMvIntegration(object):
+    """Request with retry class for TUNE Multiverse classes
+    """
+
+    _REQUEST_CONFIG = {
+        "timeout": 60,  # timeout: the number of seconds to connect to
+                        # and then read from a remote machine.
+
+        "tries": 3,  # tries: the maximum number of attempts.
+                     # (-1 is infinite). default: 10 tries.
+
+        "delay": 10  # delay: initial delay between attempts.
+                     # default: 10 seconds.
+    }
+
+    # Current directory
+    # @var string
+    CURRENT_DIR = \
+        os.path.dirname(os.path.realpath(__file__))
+
+    __session = None
+    __built_request_curl = None
+
+    @property
+    def session(self):
+        return self.__session
+
+    @session.setter
+    def session(self, value):
+        self.__session = value
+
+    @property
+    def built_request_curl(self):
+        return self.__built_request_curl
+
+    @built_request_curl.setter
+    def built_request_curl(self, value):
+        self.__built_request_curl = value
+
+    __logger = None
+
+    @property
+    def logger(
+        self
+    ):
+        """Get Property: Logger
+        """
+        if self.__logger is None:
+            self.__logger = get_logger(
+                logger_name=__name__.split('.')[0],
+                logger_version=__version__,
+                logger_format=self.logger_format,
+                logger_level=self.logger_level
+            )
+
+        return self.__logger
+
+    def __init__(
+        self,
+        logger_level=logging.INFO,
+        logger_format=TuneLoggingFormat.JSON,
+    ):
+        self.logger_level = logger_level
+        self.logger_format = logger_format
+
+        self._requests_logger()
+
+    def _requests_logger(
+        self
+    ):
+        """Set logging format to package 'requests'"""
+        if self.logger:
+            request_logger_level = self.logger_level
+
+            if request_logger_level == logging.INFO:
+                request_logger_level = logging.WARNING
+
+            tune_loggin_handler = TuneLoggingHandler(
+                logger_format=self.logger_format
+            )
+
+            tune_loggin_handler.add_logger_version(
+                'requests',
+                requests.__version__
+            )
+
+            requests_logger = logging.getLogger('requests')
+            requests_logger.addHandler(tune_loggin_handler.log_handler)
+            requests_logger.propagate = True
+            requests_logger.setLevel(level=request_logger_level)
+
+    def request(
+        self,
+        request_method,
+        request_url,
+        request_params=None,
+        request_data=None,
+        request_json=None,
+        request_retry=None,
+        request_retry_excps=None,
+        request_retry_http_status_codes=None,
+        request_retry_func=None,
```

Comment:
I would really like this notion of a retry func go away and handled as collector specific code. The HTTPAdapter Retry functionality handles the base 90% of cases.